src/pudl/metadata/resources/eiaaeo.py

```diff
@@ -0,0 +1,524 @@
+"""Table definitions for EIA AEO resources."""
+
+from typing import Any
+
+# 2024-04-24: workaround so we can *define* a bunch of schemas without
+# documenting them or expecting assets to exist
+_STAGING_RESOURCE_METADATA: dict[str, dict[str, Any]] = {
+    "core_eiaaeo__yearly_projection_electric_sector_generation_by_technology": {
+        "description": (
+            "Projected generation capacity & total generation in the electric "
+            "sector, broken out by technology."
+        ),
+        "schema": {
+            "fields": [
+                "report_year",
+                "electricity_market_module_region_aeo",
+                "modeling_case_aeo",
+                "projection_year",
+                "technology_description_aeo",
+                "summer_capacity_mw",
+                "summer_capacity_planned_additions_mw",
+                "summer_capacity_unplanned_additions_mw",
+                "summer_capacity_retirements_mw",
+                "gross_generation_mwh",
+            ],
+            "primary_key": [
+                "report_year",
+                "electricity_market_module_region_aeo",
+                "modeling_case_aeo",
+                "projection_year",
+                "technology_description_aeo",
+            ],
+        },
+        "sources": ["eiaaeo"],
+        "field_namespace": "eiaaeo",
+        "etl_group": "eiaaeo",
+    },
+    "core_eiaaeo__yearly_projection_end_use_sectors_generation_by_fuel_type": {
+        "description": (
+            "Projected generation capacity & total generation in the end-use "
+            "sector, broken out by fuel type.\n"
+            "Includes combined-heat-and-power plants and electricity-only "
+            "plants in the commercial and industrial sectors; and small on-site "
+            "generating systems in the residential, commercial, and industrial "
+            "sectors used primarily for own-use generation, but which may also "
+            "sell some power to the grid. Solar photovoltaic capacity portion of "
+            "Renewable Sources in gigawatts direct current; other technologies "
+            "in gigawatts alternating current."
+        ),
+        "schema": {
+            "fields": [
+                "report_year",
+                "electricity_market_module_region_aeo",
```

Comment:
i wonder if this guy should just be `region_eiaaeo`? that seems clear enough and distinct and less of a mouthful

Reply:
Sometimes they refer to it as "EMM region," maybe we can go with that. There are other AEO datasets that are cut up by different regions e.g. 
<img width="328" alt="image" src="https://github.com/catalyst-cooperative/pudl/assets/2495794/a0d0903e-f94b-46f8-964c-929ceac9a6db">

Reply:
By "existing name" do you mean `emm_region_eiaaeo` or `electricity_market_module_region_eiaaeo`?